app/controllers/application_templates_controller.rb

```diff
@@ -27,7 +27,24 @@ def toggle_active
   private
 
   def find_template
-    params.require :id
+    if params[:specific_path]
```

Comment:
I'm confused, isn't that exactly what I'm doing? But you need that additional parameter to be able to distinguish between WHICH route you're taking so you can specify which parameters to require. You can't just require all of them, because then both will fail to missing the other.